src/main/java/org/jenkinsci/test/acceptance/docker/DockerImage.java

```diff
@@ -92,45 +89,56 @@ public DockerImage(String tag) {
         docker.add(starter.args);
 
         File logfile = starter.log;
-        if (logfile == null) {
-            logfile = new File(cidFile + ".log");
-        }
 
-        System.out.printf("Launching Docker container `%s`: logfile is at %s\n", docker.toString(), logfile);
+        System.out.printf("Launching Docker container `%s`: logfile will be at %s\n", docker.toString(), logfile);
 
         Process p = docker.build()
                 .redirectInput(new File(SystemUtils.IS_OS_WINDOWS ? "NUL": "/dev/null"))
                 .redirectErrorStream(true)
-                .redirectOutput(logfile)
                 .start();
 
-        String cid = waitForCid(docker, cidFile, logfile, p);
+
+        String cid = waitForCid(docker, p);
+
+        Process logProcess = Docker.cmd("logs")
+                .add("-f")
+                .add(cid)
+                .build()
+                .redirectInput(new File(SystemUtils.IS_OS_WINDOWS ? "NUL": "/dev/null"))
+                .redirectErrorStream(true)
+                .redirectOutput(logfile)
+                .start();
 
         try {
             T t = type.newInstance();
-            t.init(cid, p, logfile);
+            t.init(cid, logProcess, logfile);
             return t;
         } catch (ReflectiveOperationException e) {
             throw new Error(e);
         }
     }
 
-    private String waitForCid(CommandBuilder docker, File cidFile, File logfile, Process p) throws InterruptedException, IOException {
+    private String waitForCid(CommandBuilder docker, Process p) throws InterruptedException, IOException {
         for (int i = 0; i < 10; i++) {
-            Thread.sleep(1000);
+            Thread.sleep(500);
```

Comment:
So why do you need to sleep - you now read the inputStream fully with `IOUtils.toString` - this should block until EOF is received..